mixnet/client/src/sender.rs

```diff
@@ -89,15 +98,24 @@ impl<R: Rng> Sender<R> {
     }
 
     async fn send_packet(
+        cache: &ConnectionCache,
         packet: Box<SphinxPacket>,
         addr: NodeAddressBytes,
     ) -> Result<(), Box<dyn Error>> {
         let addr = SocketAddr::try_from(NymNodeRoutingAddress::try_from(addr)?)?;
         tracing::debug!("Sending a Sphinx packet to the node: {addr:?}");
 
-        let mut socket = TcpStream::connect(addr).await?;
-        let body = Body::new_sphinx(packet);
-        body.write(&mut socket).await?;
+        if let Some(stream) = cache.get(&addr) {
+            let mut stream = stream.lock().await;
+            let body = Body::new_sphinx(packet);
+            body.write(&mut *stream).await?;
+        } else {
+            let mut socket = TcpStream::connect(addr).await?;
+            let body = Body::new_sphinx(packet);
+            body.write(&mut socket).await?;
+            cache.insert(addr, Arc::new(Mutex::new(socket)));
+        }
```

Comment:
We can probably have `cache.connect_and_get()` for example? so that we can simplify this code block.

Also, it seems there is a possibility that multiple connections to the same host can be established at the line 113, if there are multiple tasks running. This is actually not a big issue since only one will eventually survive, but this can be automatically solved if we have `cache.connect_and_get()`, I guess.